sdks/java/io/amazon-web-services/src/test/java/org/apache/beam/sdk/io/aws/sns/SnsIOTest.java

```diff
@@ -114,7 +118,10 @@ public void testDataWritesToSNS() {
 
   @Test
   public void testRetries() throws Throwable {
+    thrown.expect(IOException.class);
     thrown.expectMessage("Error writing to SNS");
+    thrown.expectMessage("No more attempts allowed");
```

Comment:
Does it expect two exception messages?

Reply:
Each `expectMessage` is just a `contains` check for the message. In this case the assertion error thrown if the logs did not exist also contained the first string :/ So the 2nd one is really the one we are looking for here as it indicates that we've given up retrying.